Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -539,41 +445,46 @@ void CGPathAddEllipseInRect(CGMutablePathRef path, const CGAffineTransform* m, C
  @Status Interoperable
 */
 void CGPathCloseSubpath(CGMutablePathRef path) {
-    if (path == NULL) {
-        return;
+    // Move the current point to the starting point since the line is closed.
+    if (!CGPointEqualToPoint(path->_impl.currentPoint, path->_impl.startingPoint)) {
+        path->_impl.currentPoint = path->_impl.startingPoint;
     }
-
-    _CGPathAddElement(path, kCGPathElementCloseSubpath);
+    path->endFigure(D2D1_FIGURE_END_CLOSED);
 }
```

Comment:
So now we are just calling endFigure and setting the current point to the starting point?

No need for the point compare, then!